dirsrvtests/tests/suites/snmp/snmp.py

```diff
@@ -0,0 +1,160 @@
+# --- BEGIN COPYRIGHT BLOCK ---
+# Copyright (C) 2024 Red Hat, Inc.
+# All rights reserved.
+#
+# License: GPL (version 3 or any later version).
+# See LICENSE for details.
+# --- END COPYRIGHT BLOCK ---
+#
+import os
+import pytest
+import logging
+import subprocess
+import ldap
+from lib389.topologies import topology_m2 as topo_m2
+
+
+DEBUGGING = os.getenv("DEBUGGING", default=False)
+if DEBUGGING:
+    logging.getLogger(__name__).setLevel(logging.DEBUG)
+else:
+    logging.getLogger(__name__).setLevel(logging.INFO)
+log = logging.getLogger(__name__)
+
+
+SNMP_USER = 'user_name'
+SNMP_PASSWORD = 'authentication_password'
+SNMP_PRIVATE = 'private_password'
+
+
+
+def run_cmd(cmd):
+    """Run a command"""
+
+    log.info(f'Run: {cmd}')
+    return subprocess.run(cmd, check=True, capture_output=True, universal_newlines=True)
+
+
+def add_lines(lines, filename):
+    """Add lines that are not already present at the end of a file"""
+
+    with open(filename, 'r') as fd:
+        for line in fd:
+            try:
+                lines.remove(line.strip())
+            except ValueError:
+                pass
+    if lines:
+        with open(filename, 'a') as fd:
+            for line in lines:
+                fd.write(f'{line}\n')
+
+
+def remove_lines(lines, filename):
+    """Remove lines in a file"""
+
+    file_lines = []
+    with open(filename, 'r') as fd:
+        for line in fd:
+            if not line.strip() in lines:
+                file_lines.append(lines)
+    with open(filename, 'w') as fd:
+        for line in file_lines:
+            fd.write(line)
+
+
+@pytest.fixture(scope="module")
+def setup_snmp(topo_m2, request):
+    """Install snmp and configure it"""
+
+    inst1 = topo_m2.ms["supplier1"]
+    if not inst1.with_systemd_running():
+        pytest.skip('This test requires systemd')
+        return None
+
+    # Install snmp
+    run_cmd(['/usr/bin/dnf', 'install', '-y', 'net-snmp', 'net-snmp-utils', '389-ds-base-snmp'])
```

Comment:
We should not install anything during the tests, instead we should check if the required executables are present, and skip the test if they are missing. See for example https://github.com/389ds/389-ds-base/blob/c019af140ca53955e67038e1c9fe18620410638d/dirsrvtests/tests/suites/basic/basic_test.py#L1230-L1231